DRIVERS-2878 OIDC Atlas Testing Updates (#1558)
mongodb/specifications: source/auth/tests/mongodb-oidc.md

> For test cases that create fail points, drivers MUST either use a unique `appName` or explicitly remove the fail point
> callback to prevent interaction between test cases.

After setting up your OIDC
[environment](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/auth_oidc/README.md),
source the `secrets-export.sh` file and use the associated env variables in your tests.

> The default client should set the appropriate `ENVIRONMENT` auth mechanism property and include a callback that gets the

Reviewer: ```suggestion
An OIDC configured client MUST set the appropriate `ENVIRONMENT` auth mechanism property and include a callback that gets the
```
To clarify the term "OIDC configured client", which is used later?

Drivers that support the [Human Authentication Flow](../auth.md#human-authentication-flow) MUST implement all prose
tests in this section. Unless otherwise noted, all `MongoClient` instances MUST be configured with `retryReads=false`.

The human workflow tests MUST only be run when in `ENVIRONMENT:test`.

> [!NOTE]
> For test cases that create fail points, drivers MUST either use a unique `appName` or explicitly remove the fail point
> after the test to prevent interaction between test cases.

Drivers MUST be able to authenticate against a server configured with either one or two configured identity providers.

Unless otherwise specified, use `MONGODB_URI_SINGLE` and the `test_user1` token in the `OIDC_TOKEN_DIR` as the
"access_token", and a dummy "refresh_token" for all tests.

When using an explicit username for the client, we use the token name and the domain name given by `OIDC_DOMAIN`, e.g.
`test_user1@${OIDC_DOMAIN}`.

### (1) OIDC Human Callback Authentication

Drivers MUST be able to authenticate using OIDC callback(s) when there is one principal configured.

**1.1 Single Principal Implicit Username**

> - Create default client

Reviewer: Is it correct that a "default client" is the same as an "OIDC configured client"? If so, suggest using the more specific "OIDC configured client" term throughout, including the "create a client" statements below.

Author: Updated for consistency.
